Data/Xml/TroopSpellBookData.cs

```diff
@@ -0,0 +1,52 @@
+﻿using EOAE_Code.Data.Managers;
+using EOAE_Code.Magic.Spells;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using TaleWorlds.Core;
+
+namespace EOAE_Code.Data.Xml
+{
+    [Serializable]
+    public class TroopSpellBookData
+    {
+        [XmlAttribute]
+        public string TroopId = "";
+        [XmlArray]
+        public TroopSpellData[] Spells = new TroopSpellData[0];
+
+        public void Normalize()
+        {
+            float totalWeight = 0;
+
+            for (int i = 0; i < Spells.Length; i++)
+            {
+                totalWeight += Spells[i].Weight;
+            }
```

Comment:
This could also be done with LINQ:
```suggestion
            var totalWeight = Spells.Sum(it => it.Weight);
```